File: csharp/resistor-color/ResistorColor.cs
using System;
public static class ResistorColor{
    public static int ColorCode(string color){
        String[] a = new String[10];
        a = Colors();
        int pos = Array.IndexOf(a, color);
        return pos;
    }

    public static string[] Colors(){
        String[] arr = new String[] {"black","brown","red","orange","yellow","green","blue","violet","grey","white"};
        return arr;
    }
}

File: java/resistor-color/src/main/java/ResistorColor.java

import java.util.*;
class ResistorColor {
     int colorCode(String color) {
        for (int i = 0; i < colors().length; i++) {
            if (colors()[i].equals(color)){
                return i;
            }
        }
        return -1;
    }
    String[] colors() {
        String[] arr = new String[] {"black","brown","red","orange","yellow","green","blue","violet","grey","white"};
        return arr;
    }
}
